ios/FluentUI/Tab Bar/SideTabBar.swift

```diff
@@ -0,0 +1,286 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+/// Delegate protocol to handle user events inside the side tab bar.
+@objc(MSFSideTabBarDelegate)
+public protocol SideTabBarDelegate {
+    /// Called after the view representing `TabBarItem` is selected.
+    /// - Parameter sideTabBar: The side tab bar.
+    /// - Parameter item: The selected tab bar item.
+    /// - Parameter fromTop: true if the item was in the top section, false if it was in the bottom section.
+    @objc optional func sideTabBar(_ sideTabBar: SideTabBar, didSelect item: TabBarItem, fromTop: Bool)
+
+    /// Called after the avatar view is tapped in the side tab bar.
+    /// - Parameter sideTabBar: The side tab bar.
+    /// - Parameter avatarView: The avatar view.
+    @objc optional func sideTabBar(_ sideTabBar: SideTabBar, didActivate avatarView: AvatarView)
+}
+
+/// View for a vertical side tab bar that can be used for app navigation.
+/// Optimized for horizontal regular + vertical regular size class configuration. Prefer using TabBarView for other size class configurations.
+@objc(MSFSideTabBar)
+open class SideTabBar: UIView {
+    /// Delegate to handle user interactions in the side tab bar.
+    @objc public weak var delegate: SideTabBarDelegate?
+
+    /// The avatar view that displays above the top tab bar items.
+    /// The avatar view's size class should be AvatarSize.medium.
+    @objc open var avatarView: AvatarView? {
```

Comment:
i am not sure we should expose avatarView if we are saying we going to restrict the size as "medium"

Reply:
I am also going to warn you that medium avatar size is going to change soon from 30 to 32.

Reply:
Fortunately, we can enforce that the avatar view's size is .medium since it's a readwrite property.
I like that we are giving the flexibility of passing in an AvatarView because there are many ways to configure the avatar view. Otherwise, we would need to create a mirror interface that pretty much matches the AvatarView public interface and then internally create an AvatarView based off of that. I think this would make the code more complex and less maintainable. With this approach, we are providing more flexibility, without giving too much flexibility since we are force the view to be an AvatarView and not any random UIView.